### PR TITLE
Integrate SAM into cellprofiler and post-processing for image-based profiles

### PR DESCRIPTION
**Pull request by MikeLippincott**
This PR is for a subset of data to test the object tracking of SAM and merge coordinates with cellprofiler outputs. The next PR will run the full dataset.

### CODE REVIEW

File: 5.process_CP_features/scripts/2.combine_terminal_data.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+# In[8]:
+
+
+terminal_df.head()
+
+
+# In[ ]:
+
+
+# In[ ]:
```
Comment (jenna-tomkinson) [inline 5.process_CP_features/scripts/2.combine_terminal_data.py:129]: Make sure to remove these if not needed.

File: 2.cellprofiler_ic_processing/scripts/1.process_ic_teminal_data.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+# number of timepoints in the main data
+num_timepoints = 13
+
+
+# get the list of files in the terminal data directory that are tiffs
+tiff_files = list(terminal_data_path.glob("*.tiff"))
+tiff_files = sorted(tiff_files)
+# change the timepoint from "T0001" to "T0014" to match the main data format and position
+# tiff_files = [str(file).replace("T0001", "T0014") for file in tiff_files]
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 2.cellprofiler_ic_processing/scripts/1.process_ic_teminal_data.py:47]: removed!

File: 2.cellprofiler_ic_processing/scripts/perform_ic.py
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 2.cellprofiler_ic_processing/scripts/perform_ic.py:1]: removed

File: 3b.run_sam/scripts/1.run_sam2_microscopy.py [hunk 3]
```diff
@@ -224,7 +235,7 @@
 
 # need to downscale to fit the model and images on the GPU
 # note that this is an arbitrary number and can be changed
-downscale_factor = 8
+downscale_factor = 5
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 3b.run_sam/scripts/1.run_sam2_microscopy.py:238]: less downscale = more accurate segmentation, but also more GPU VRAM...its a trade off so I have been changing it

File: 4.cellprofiler_analysis/scripts/run_cellprofiler_analysis.py [hunk 4]
```diff
@@ -15,6 +15,7 @@
 sys.path.append("../../utils/")
 import cp_parallel
 
+
 # ## Set paths and variables
 
 # In[ ]:
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 4.cellprofiler_analysis/scripts/run_cellprofiler_analysis.py:21]: correct, not ran

File: 5.process_CP_features/scripts/1.annotate_sc.py [hunk 5]
```diff
@@ -11,13 +11,103 @@
...
+    pd.DataFrame
+        A merged dataframe of the two input dataframes based on the euclidean distance between the coordinates
+    """
+    # split the each data fram into each cell_merge_column
+    all_images = df_left[unique_image_column].unique()
+
+    merged_df_list = []
+    total_CP_cells = 0
+    total_annotated_cells = 0
+    distances = []
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 5.process_CP_features/scripts/1.annotate_sc.py:64]: added comments!

File: 5.process_CP_features/scripts/1.annotate_sc.py [hunk 6]
```diff
@@ -102,132 +201,211 @@
...
+        if not col.startswith("Metadata_"):
+            location_metadata_df = location_metadata_df.rename(
+                columns={col: f"Metadata_{col}"}
+            )
+
+    ##############################################
+    if "endpoint" in data_run:
+        # get only the endpoint timepoint for the endpoint data
+        location_metadata_df = location_metadata_df[
+            location_metadata_df["Metadata_Time"] == "0014"
```
[the remark this answers is not in the document]
Comment (MikeLippincott) [reply 5.process_CP_features/scripts/1.annotate_sc.py:256]: I think it is four total placeholders

**Pull request merged**